backend/src/connectors/peer/huggingface.ts

```
@@ -0,0 +1,106 @@
+    if (!extra || !Object.hasOwn(extra, 'statusModelId') || !Object.hasOwn(extra, 'statusModelName')) {
+      throw ConfigurationError(`Missing configuration for HuggingFaceHub ${this.id}`)
+    }
+    return this.config.extra as ExtraConfig
+  }
+
+  init(): Promise<boolean> {
+    this.getExtraConfig()
+    return Promise.resolve(true)
+  }
+
+  async getPeerStatus(): Promise<SystemStatus> {
+    try {
+      // There isn't a general 'status' endpoint, so check a 'known' model
+      const name = this.getExtraConfig().statusModelName
+      const modelId = this.getExtraConfig().statusModelId
+      const model = await modelInfo({ name })
+      if (model.id !== modelId) {
+        return {
+          ping: 'pong',
+          error: {
+            status: 500,
+            code: 500,
+            name: 'HuggingFaceHub API error',
+            message: 'Unexpected ID returned when checking known model on HuggingFaceHub',
+            context: { model },
+          },
+        }
+      }
+    } catch (error) {
+      if (error instanceof HubApiError) {
+        return {
+          ping: '',
+          error: {
+            ...error,
+            status: error.statusCode,
+            code: error.statusCode,
+          },
+        }
+      }
+    }
+
+    return {
+      ping: 'pong',
+    }
+  }
+
+  async queryModels(opts: { query: string }, user: UserInterface): Promise<Array<ModelSearchResult>> {
+    const cache = this.getQueryCache()
+    const cacheKey = this.buildCacheKey(user, opts.query)
+    const cacheResult = cache ? cache.get<Array<ModelSearchResult>>(cacheKey) : undefined
+    if (cacheResult) {
+      log.debug(`Cache hit - returning cached result for ${cacheKey}`)
+      return cacheResult
+    }
+    log.debug(`Cache miss - executing query for ${cacheKey}`)
+
+    // No cache result - execute query, cache response if appropriate
+    const models = new Array<ModelSearchResult>()
+    // Don't allow a whole search
+    if (opts.query.length < 5) {
```

I don't like this returning "silently" - can we throw an error instead? This way the frontend can handle the error and let the user know why the result is not what they might otherwise be expecting.